src/Microsoft.AspNetCore.SignalR.Core/DefaultHubLifetimeManager.cs

```diff
@@ -202,18 +211,30 @@ public override Task SendGroupExceptAsync(string groupName, string methodName, o
             var group = _groups[groupName];
             if (group != null)
             {
-                var message = CreateInvocationMessage(methodName, args);
-                var tasks = group.Values.Where(connection => !excludedIds.Contains(connection.ConnectionId))
-                    .Select(c => c.WriteAsync(message).AsTask());
+                var message = CreateSerializedInvocationMessage(methodName, args);
+                var tasks = SendMessageToConnections(message, group.Values.Where(connection => !excludedIds.Contains(connection.ConnectionId)));
                 return Task.WhenAll(tasks);
             }
 
             return Task.CompletedTask;
         }
 
-        private SerializedHubMessage CreateInvocationMessage(string methodName, object[] args)
+        private IEnumerable<Task> SendMessageToConnections(SerializedHubMessage hubMessage, IEnumerable<HubConnectionContext> connectionContexts)
         {
-            return new SerializedHubMessage(new InvocationMessage(target: methodName, argumentBindingException: null, arguments: args));
+            foreach (var connectionContext in connectionContexts)
+            {
+                yield return connectionContext.WriteAsync(hubMessage).AsTask();
+            }
+        }
+
+        private SerializedHubMessage CreateSerializedInvocationMessage(string methodName, object[] args)
+        {
+            return new SerializedHubMessage(CreateInvocationMessage(methodName, args));
+        }
+
+        private HubMessage CreateInvocationMessage(string methodName, object[] args)
+        {
+            return new InvocationMessage(target: methodName, argumentBindingException: null, arguments: args);
```

Comment:
Off topic: As far as I can tell the `params` on `InvocationMessage` is doing the opposite of saving keystrokes. It is never used and it forces named arguments to be used whenever constructed. Any objections to removing `params` in a followup PR?

Reply:
Remove them, I saw them and I feel like they were added for the benchmarks https://github.com/aspnet/SignalR/blob/0f663cadc4a2d28de96a92e1abc3cba5ac2b85b9/benchmarks/Microsoft.AspNetCore.SignalR.Microbenchmarks/HubProtocolBenchmark.cs#L42-L50